MapboxNavigation/RouteVoiceController.swift

```diff
@@ -57,14 +57,19 @@ open class RouteVoiceController: NSObject, AVSpeechSynthesizerDelegate, AVAudioP
     public var bufferBetweenAnnouncements: TimeInterval = 3
     
     /**
+     Delegate used for getting metadata information about a particular spoken instruction.
+     */
+    public weak var voiceControllerDelegate: VoiceControllerDelegate?
+    
+    /**
      Default initializer for `RouteVoiceController`.
      */
     override public init() {
         super.init()
         
         if !Bundle.main.backgroundModes.contains("audio") {
-            print("Voice guidance may not work properly. " +
-                  "Add audio to the UIBackgroundModes key to your app’s Info.plist file")
+            let errorMessage = "Voice guidance may not work properly. Add audio to the UIBackgroundModes key to your app’s Info.plist file"
+            voiceControllerDelegate?.spokenInstructionsDidFail?(self, error: NSError(localizedFailureReason: "Unable to read instruction aloud", detailedFailureReason: errorMessage, code: .audioBackgroundModeMissing)) ?? print(errorMessage)
```

Comment:
I think we should make this flag a hard requirement for the navigation SDK and maybe have it fail earlier, when creating the NavigationViewController. At least make it an assertion, because the user should be able to expect to hear the same spoken instructions while the application is in the background.